test/session.test.js

```diff
@@ -676,3 +677,54 @@ test('save supports rejecting promises', async t => {
   // 200 since we assert inline and swallow the error
   t.is(response.statusCode, 200)
 })
+
+test('only save session when it changes', async t => {
+  t.plan(6)
+  const setStub = sinon.stub()
+  const store = new Map()
+
+  const fastify = Fastify()
+  fastify.register(fastifyCookie)
+
+  fastify.register(fastifySession, {
+    ...DEFAULT_OPTIONS,
+    saveUninitialized: false,
+    cookie: { secure: false },
+    store: {
+      set (id, data, cb) {
+        setStub()
+        store.set(id, data)
+        cb(null)
+      },
+      get (id, cb) { cb(null, store.get(id)) },
+      destroy (id, cb) {
+        store.delete(id)
+        cb(null)
+      }
+    }
+  })
+
+  fastify.get('/', (request, reply) => {
+    request.session.userId = 42
+
+    reply.send(200)
+  })
+
+  const { statusCode: statusCode1, headers: headers1 } = await fastify.inject('/')
```

Comment:
I assume there are legacy reasons (i.e., the tests came with the fork) for not using `inject` in the other tests and this is fine to use?